src/include/common/data_chunk/sel_vector.h

```diff
@@ -21,7 +22,11 @@ class SelectionVector {
 
     SelectionVector() : SelectionVector{DEFAULT_VECTOR_CAPACITY} {}
 
-    bool isUnfiltered() const { return selectedPositions == (sel_t*)&INCREMENTAL_SELECTED_POS; }
+    bool isUnfiltered() const { return selectedPositions == INCREMENTAL_SELECTED_POS.data(); }
+    bool isContiguous() const {
+        return selectedPositions >= &*INCREMENTAL_SELECTED_POS.begin() &&
+               selectedPositions <= &*INCREMENTAL_SELECTED_POS.end();
+    }
 
     void setToUnfiltered() { selectedPositions = (sel_t*)&INCREMENTAL_SELECTED_POS; }
```

Comment:
Since you're changing this class, can u also refactor away the c-style castings?

Reply:
I guess so. It would make the const_cast more explicit, but I also don't really like that we're const-casting here. Admittedly it will segfault if you attempt to use `operator[]` to modify it when selectedPositions is pointing at `INCREMENTAL_SELECTED_POS`.